// Protobuf 语法的版本
syntax = "proto3";

// 定义包名，便于在生成的代码中区分不同模块（类似 C++ 的命名空间）
package user;

// 允许生成通用的 C++ 服务接口（可选项）
option cc_generic_services = true;

message ResultCode {
    int32 errcode = 1; 
    bytes errmsg = 2;
}

message LoginRequest {
    bytes name = 1;
    bytes password = 2;
}

message LoginResponse {
    ResultCode result = 1;
    bool success = 2;
}

message RegisterRequest {
    uint32 id = 1;
    bytes name = 2;
    bytes password = 3;
}

message RegisterResponse {
    ResultCode result = 1;
    bool success = 2;
}

// 定义RPC服务接口类和服务函数
service UserServiceRpc {
    rpc Login(LoginRequest) returns (LoginResponse);

    rpc Register(RegisterRequest) returns (RegisterResponse);
}
